kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -290,14 +300,24 @@ impl GotocHook for PointerOffset {
     fn handle(
         &self,
         gcx: &mut GotocCtx,
-        _instance: Instance,
+        instance: Instance,
         mut fargs: Vec<Expr>,
         assign_to: &Place,
         target: Option<BasicBlockIdx>,
         span: Span,
     ) -> Stmt {
         assert_eq!(fargs.len(), 1);
-        let ptr = fargs.pop().unwrap().cast_to(Type::void_pointer());
+        let ptr = {
+            let ptr = fargs.pop().unwrap();
+            let place_ty = instance.args().0[0].expect_ty().clone();
+            // Handle both fat and thin pointers.
+            if gcx.use_thin_pointer_stable(place_ty) {
+                ptr
+            } else {
+                ptr.member("data", &gcx.symbol_table)
+            }
+        }
+        .cast_to(Type::void_pointer());
```

Comment:
can we do that as part of the kani library implementation using `to_raw_parts()`?